src/OracleService/Protocols/OracleFsProtocol.cs

```diff
@@ -0,0 +1,141 @@
+using Google.Protobuf;
+using Neo.Network.P2P.Payloads;
+using NeoFS.API.v2.Client;
+using NeoFS.API.v2.Client.ObjectParams;
+using NeoFS.API.v2.Refs;
+using NeoFS.API.v2.Cryptography;
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Range = NeoFS.API.v2.Object.Range;
+using System.Collections.Generic;
+using Object = NeoFS.API.v2.Object.Object;
+using Neo.Wallets;
+using Neo.Cryptography.ECC;
+
+namespace Neo.Plugins
+{
+    class OracleFsProtocol : IOracleProtocol
+    {
+        private byte[] privateKey;
+        private const string URIScheme = "neofs";
+        private const string RangeCmd = "range";
+        private const string HeaderCmd = "header";
+        private const string HashCmd = "hash";
+
+        public void Configure()
+        {
+        }
+
+        public void AttachWallet(Wallet wallet, ECPoint[] oracles)
+        {
+            privateKey = oracles.Select(p => wallet.GetAccount(p)).Where(p => p is not null && p.HasKey && !p.Lock).FirstOrDefault()?.GetKey().PrivateKey;
+        }
+
+        public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, CancellationToken cancellation)
+        {
+            Utility.Log(nameof(OracleFsProtocol), LogLevel.Debug, $"Request: {uri.AbsoluteUri}");
+
+            if (!Settings.Default.AllowPrivateHost)
+            {
+                IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host);
+                if (entry.IsInternal())
+                    return (OracleResponseCode.Forbidden, null);
+            }
+            int index = uri.LocalPath.GetHashCode() % Settings.Default.Fs.FSNodes.Length;
```

Comment:
`GetHashCode` is not deterministic, is pretended to be a random index?

Reply:
Yes.A random index is needed to select the requested node.

Reply:
Then I prefer to use `Random`